internal/api/v1/handlers/user.go

```diff
@@ -0,0 +1,102 @@
+package handlers
+
+import (
+	"errors"
+
+	"github.com/gofiber/fiber/v2"
+	"gorm.io/gorm"
+
+	"github.com/celestiaorg/talis/internal/api/v1/services"
+	"github.com/celestiaorg/talis/internal/types/infrastructure"
+)
+
+// UserHandler handles HTTP requests for user operations
+type UserHandler struct {
+	service *services.User
+}
+
+// NewUserHandler creates a new UserHandler instance
+func NewUserHandler(service *services.User) *UserHandler {
+	return &UserHandler{
+		service: service,
+	}
+}
+
+// Define common errors
+var (
+	ErrInvalidUserID      = errors.New("invalid user id")
+	ErrInvalidUsername    = errors.New("invalid username")
+	ErrUserNotFoundByID   = errors.New("user not found with provided id")
+	ErrUserNotFoundByName = errors.New("user not found with provided username")
+)
+
+// CreateUser handles the creation of a new user
+func (h *UserHandler) CreateUser(c *fiber.Ctx) error {
+	var userReq infrastructure.CreateUserRequest
+	if err := c.BodyParser(&userReq); err != nil {
+		return c.Status(fiber.StatusBadRequest).
+			JSON(infrastructure.ErrInvalidInput(err.Error()))
+	}
+
+	if err := userReq.Validate(); err != nil {
+		return c.Status(fiber.StatusBadRequest).
+			JSON(infrastructure.ErrInvalidInput(err.Error()))
+	}
+
```

Comment:
> This request should be validated somewhere. I think the best place would be at the `repo/user.go` level.
> 
> To start it should be verifying that the username name is not empty.

@MSevey  Would improving this validation be enough? 
The validation covers an empty username field and can be extended to validate other fields